orchagent/orchdaemon.cpp

```diff
@@ -248,13 +248,26 @@ bool OrchDaemon::init()
 
         static const vector<sai_queue_attr_t> queueAttrIds;
 
-        m_orchList.push_back(new PfcWdSwOrch<PfcWdZeroBufferHandler, PfcWdLossyHandler>(
-                    m_configDb,
-                    pfc_wd_tables,
-                    portStatIds,
-                    queueStatIds,
-                    queueAttrIds,
-                    PFC_WD_POLL_MSECS));
+        if (platform == MLNX_PLATFORM_SUBSTRING)
+        {
+            m_orchList.push_back(new PfcWdSwOrch<PfcWdZeroBufferHandler, PfcWdLossyHandler>(
+                        m_configDb,
+                        pfc_wd_tables,
+                        portStatIds,
+                        queueStatIds,
+                        queueAttrIds,
+                        PFC_WD_POLL_MSECS));
+        }
+        else if (platform == BFN_PLATFORM_SUBSTRING)
+        {
+            m_orchList.push_back(new PfcWdSwOrch<PfcWdAclHandler, PfcWdLossyHandler>(
```

Comment:
Are you sure bfn wants to take the ACL path? One head-up for you is that the ACL entry for PFC WD by default does not have ACL counters attached to it. If the drop stats, which are fetched by pfcwd show stats, are obtained from ACL counters on bfn, we may have a problem, and we may need to rethink the landscape to embrace both.

Reply:
> Are you sure bfn wants to take the ACL path? One head-up for you is that the ACL entry for PFC WD by default does not have ACL counters attached to it. If the drop stats, which are fetched by pfcwd show stats, are obtained from ACL counters on bfn, we may have a problem, and we may need to rethink the landscape to embrace both.

Yes, we've taken this into account and this approach works for us. The pfc_wd CT is passing for our platform with ACL drop handler.